src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java

```diff
@@ -215,13 +215,29 @@ private void trySubmit(@NotNull Project project)
     groups.add(0, new Group(-1, Collections
         .singletonList(getText("ui.toolWindow.subTab.exercises.submission.submitAlone"))));
 
-    SubmissionViewModel submission =
-        new SubmissionViewModel(exercise, submissionInfo, history, groups, files, language);
+    // Find the group from the available groups that matches the default group ID.
+    // A group could be removed, so this way we check that the default group ID is still valid.
+    Optional<Long> defaultGroupId = PluginSettings.getInstance().getDefaultGroupId();
+    Group defaultGroup = defaultGroupId
+        .flatMap(id -> groups
+            .stream()
+            .filter(group -> group.getId() == id)
+            .findFirst())
+        .orElse(null);
+
+    SubmissionViewModel submission = new SubmissionViewModel(
+        exercise, submissionInfo, history, groups, defaultGroup, files, language);
 
     if (!dialogs.create(submission, project).showAndGet()) {
       return;
     }
 
+    if (Boolean.TRUE.equals(submission.makeDefaultGroup.get())) {
+      PluginSettings.getInstance().setDefaultGroupId(submission.selectedGroup.get().getId());
+    } else {
+      PluginSettings.getInstance().clearDefaultGroupId();
```

Comment:
Or should the default just to stay being what it was.

Reply:
That's a good point. So it could be used to make another group the default value, but "undefaulting" a group requires a settings reset.

My only concern is whether it's unintuitive that unchecking it actually has no effect once you already have a default group

Reply:
Perhaps the correct solution here is disable the checkbox if the selected group is the default group?

Reply:
I think the current way is ok.

Disabling checkbox could be a bit confusing for users even though it would kinda make sense to me.

More complex logic I thought is that when group selection is changed by dropdown list, the status of the checkbox auto-updares to reflect whether or not that is the current default.  If user doesn't change that, the default remains. If user unchecks automatically checked checkbox, default is reset. If user checks auto-unchecked checkbox, default is changed to be the new value.

IF selected == default XOR makeDefaultChecked THEN default = makeDefaultChecked ? selected : null

Reply:
So just to make sure I understood what you suggested correctly: otherwise equivalent to the current solution (unchecking resets), but selecting a non-default group automatically unchecks the check box?

Reply:
Almost just as you said but with an additional rule that if the selected group is not the default group and the checkbox is unchecked (which is automatically is after selecting that group from dropdown list), then the default group is not changed nor cleared (it keeps the same).

But I see now that this is quite confusing logic too. Perhaps just go with what it is now. That's good and sensible, after all.

Reply:
Sounds good, will keep it as is so it's free of too much complexity